bluesky/plans.py

```diff
@@ -90,36 +96,67 @@ def list_scan(detectors, motor, steps, *, per_step=None, md=None):
     --------
     :func:`bluesky.plans.rel_list_scan`
     """
+    if len(args) % 2 != 0:
+            raise ValueError("The list of arguments must contain a list of"
+                             "points for each defined motor")
+
+    # check that all lists are the same length
+    lengths = {}
+    motors = []
+    pos_lists = []
+    for motor, pos_list in partition(2,args):
+        lengths[motor.name] = len(pos_list)
+        motors.append(motor)
+        pos_lists.append(pos_list)
+    lengths = {motor.name: len(pos_list) for motor, pos_list
+               in partition(2,args)}
+    length_check = all(elem == list(lengths.values()[0]) for elem in
+                       list(lengths.values()))
+
+    if not length_check:
+        raise ValueError("The lengths of all lists in *args must be the same"
+                         f"However the lengths in args are : {lengths}")
+
```

Comment:
At this point, where we know we have one consistent length, it might be worth defining a variable `length = len(post_lists([0])` to improve readability, since you use this expression several places below.

Reply:
done, good call